sanic/blueprints.py

```diff
@@ -337,10 +337,11 @@ def register(self, app, options):
                 middleware.append(app._apply_middleware(future, route_names))
 
         # Exceptions
-        for future in self._future_exceptions:
-            exception_handlers.append(
-                app._apply_exception_handler(future, route_names)
-            )
+        if route_names:
```

Comment:
Just being a bit pedantic here, but you should be able to remove this `if` statement only and it should still work (since the code is now indented inside another `if route_names` check :)

Reply:
You're right. I was thinking it would be cleaner this way since each primitive we are applying is separate.